derive_client/data_types/models.py

```diff
@@ -66,3 +67,29 @@ class NonMintableTokenData(TokenData):
 class DeriveAddresses(BaseModel):
     model_config = ConfigDict(arbitrary_types_allowed=True)
     chains: dict[ChainID, dict[Currency, MintableTokenData | NonMintableTokenData]]
+
+
+@dataclass
+class TxResult:
+    tx_hash: str
+    tx_receipt: AttributeDict | None
+    exception: Exception | None
+
+    @property
+    def status(self) -> TxStatus:
+        if self.tx_receipt is not None:
+            return TxStatus(int(self.tx_receipt.status))  # ∈ {0, 1} (EIP-658)
+        if isinstance(self.exception, TimeoutError):
+            return TxStatus.PENDING
+        return TxStatus.ERROR
+
+
+@dataclass
+class BridgeResult:
+    source_chain: ChainID
+    target_chain: ChainID
+    source_token: Currency
+    target_token: Currency
+    amount: int
+    receiver: Address
+    tx_result: TxResult
```

Comment:
Keeping `BridgeResult` as a pure data container means we can cleanly add a method on `BaseClient` like `check_bridge_result()` to poll the latest tx status. This avoids forcing the caller to handle Web3 setup or determine RPC endpoints themselves.